mailet/api/src/main/java/org/apache/mailet/Serializer.java

```diff
@@ -266,6 +268,30 @@ public boolean equals(Object other) {
 
     Serializer<Double> DOUBLE_SERIALIZER = new DoubleSerializer();
 
+    class ByteArraySerializer implements Serializer<BytesArrayDto> {
+        @Override
+        public JsonNode serialize(BytesArrayDto object) {
+            return STRING_SERIALIZER.serialize(Base64.getEncoder().encodeToString(object.getValues()));
```

Comment:
Two concerns:
1. I've just checked `encodeToString()` method, and this is the documentation:
```java
         * <p> This method first encodes all input bytes into a base64 encoded
         * byte array and then constructs a new String by using the encoded byte
         * array and the {@link java.nio.charset.StandardCharsets#ISO_8859_1
         * ISO-8859-1} charset.
         *
         * <p> In other words, an invocation of this method has exactly the same
         * effect as invoking
         * {@code new String(encode(src), StandardCharsets.ISO_8859_1)}.
```

It transforms a byte[] to `base64` byte[], then create new String from `based64` byte[] in `ISO_8859_1` charset. Do we want to use `UTF-8` charset? And if yes, it should be:
```java
return STRING_SERIALIZER.serialize(new String(Base64.getEncoder().encode(object.getValues()), StandardCharsets.UTF_8));
```
Similar to deserialization
2. Why not transform directly a byte[] to a String?
```java
return STRING_SERIALIZER.serialize(new String(object.getValues(), StandardCharsets.UTF_8));
```

Reply:
1. Nice catch
2. Because UTF-8 Can not encore every bytes combinations:

```
        @Test
        void byteArrayShouldBeSerializedAndBack() {
            List<Byte> base = IntStream.rangeClosed(Byte.MIN_VALUE, Byte.MAX_VALUE).mapToObj(Integer::toString).map(Byte::parseByte).collect(Collectors.toList());
            base.forEach(b1 ->
                    base.forEach(b2 ->
                            base.forEach(b3 ->
                                base.forEach(b4 -> {
                                    AttributeValue<BytesArrayDto> expected = AttributeValue.of(new BytesArrayDto(new byte[] { b1, b2, b3, b4 }));

                                    JsonNode json = expected.toJson();
                                    AttributeValue<?> actual = AttributeValue.fromJson(json);

                                    assertThat(((BytesArrayDto) actual.getValue()).getValues()).isEqualTo(expected.getValue().getValues());
                                }))));
        }
```